artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerSenderContext.java

```diff
@@ -733,20 +734,31 @@ private static boolean hasCapabilities(Symbol symbol, Source source) {
       return false;
    }
 
-   private static String createQueueName(String clientId,
+   private static String createQueueName(boolean useLegacyQueueNaming,
+                                         String clientId,
                                          String pubId,
                                          boolean shared,
                                          boolean global,
                                          boolean isVolatile) {
-      String queue = clientId == null || clientId.isEmpty() || global ? pubId : clientId + "." + pubId;
-      if (shared) {
-         if (queue.contains("|")) {
-            queue = queue.split("\\|")[0];
-         }
-         if (isVolatile) {
-            queue = "nonDurable" + "." + queue;
+      if (useLegacyQueueNaming) {
+         String queue = clientId == null || clientId.isEmpty() ? pubId : clientId + "." + pubId;
+         if (shared) {
+            if (queue.contains("|")) {
+               queue = queue.split("\\|")[0];
+            }
+            if (isVolatile) {
+               queue += ":shared-volatile";
+            }
+            if (global) {
+               queue += ":global";
+            }
          }
+         return queue;
+      } else {
+         final boolean durable = !isVolatile;
+         final String subscriptionName = pubId.contains("|") ? pubId.split("\\|")[0] : pubId;
+         final String clientID = clientId == null || clientId.isEmpty() || global || pubId.endsWith("|global") ? null : clientId;
```

Comment:
@gemmellr it seems when you do a JMS unsubscribe from the AMQP client, the source is null as such don't have this field, if you could point me to where else we could gain this info in a better fashion?